### DISCUSSION TITLE
Using Root based formulas

### DISCUSSION THREAD

**Discussion by alefisico** [cms-nanoAOD/correctionlib#228] [Q&A] [answered]
Hi,

I have a ROOT-based formula from TFitResults, which I want to use correctionlib to evaluate. My function is like:
```
ROOT::Math::crystalball_cdf(x,[p5],[p4],[p1],[p0])*(0.5*(1+TMath::Erf((x-[p0])/[p6])))*([p3]-[p2])+[p2] 
```
Is it possible to use correctionlib for this?

**Comment by nsmith-** [comment, marked as answer]
Not without extending the parser grammar: we don't want to adopt a dependency on ROOT in this library and TFormula is not thread-safe. See https://github.com/cms-nanoAOD/correctionlib/discussions/17 for some more motivation and history.
That said, there might be good motivation to extend the parser and/or merge efforts with formulate. See https://github.com/scikit-hep/formulate/issues/55 for more details.
